### ISSUE TITLE
Spaces within headings in HTML

### ISSUE THREAD

**Issue by rbeezer** [rbeezer/mathbook#897]
Full headings in HTML often go something like this, but all on one line, with no spaces:  

```html
<h2 class="heading hide-type">
  <span class="type">Section</span>
  <span class="codenumber">11</span>
  <span class="title">Further Reading</span>
</h2>
```

The absence of spaces makes screen readers run all the words together, and if CSS fails to load (accidentally or due to restrictions), then titles will be "smashed" together.

It is easy enough to edit the necessary HTML if we want to do this.  It might take some thought to put them in the right places so that the `.hide-type` class will do the right thing.

Opening the issue for discussion, and likely coordination of the release of HTML and CSS.

**Comment by rbeezer** [comment]
On 05/29/2018 11:37 AM, David W. Farmer wrote:
> There are many places where have no spaces inside an hN,
> such as theorem-like.

Right.  By "full" I meant to include these.

> I'd like to check that the punctuation added by CSS will
> still work properly.

I can do a cursory implementation and make a "beta" - should we proceed that way?

Rob

**Comment by rbeezer** [comment]
I've inserted spaces within the headings every place I could think of.  Notes:

* `figcaption` is a place to look.  On `sidebyside` panels these are causing validation errors, so don't work too hard on them, they may change soon.  (I'll consult.)
* Similarly, look at "nav links" on summary pages.  We are suppose to redo those also, so don't sweat them yet?
* "backmatter" exercises will definitely be re-done soon.  Don't work too hard.
* I detect **no visual differences** in the sample article.  Suggesting we could release this sooner rather than later and roll-out fixes and CSS tweaks as needed.
* I've tested source and rendering changes carefully for sample article.  Sample book is produced blindly.
* Not sure if `heading.hide-type` would suggest we should bury a space **within** the span, rather than between.

http://mathbook.pugetsound.edu/beta/20180531-article-spaces/

http://mathbook.pugetsound.edu/beta/20180531-book-spaces/

**Comment by rbeezer** [comment]
Merged at d19ad4ea15f9e1b334df8b64de5a334309395658 with a temporary title I failed to edit.  So it may end up being harder to find.  I'll leave this open in case we need to discuss fall-out.
